### PR TITLE
`.xmp` support in license attribution

### LINKED ISSUE #197: Offer XMP metadata file for download
## Problem

Currently, there's no way to generate a XMP metadata file with the chooser.

## Description

Although I do not consider myself a novice to the Creative Commons licenses, I used the legacy license chooser a lot to create proper machine readable data until I wrote my own metadata generator. So I believe it might facilitate the work of other people interested in marking other kinds of work (especially PDFs) to offer a XMP metadata file.

## Alternatives

Only linking to the [XMP](https://wiki.creativecommons.org/wiki/XMP) entry on the wiki. It doesn't even contain a complete example right now, so not a good option in my opinion. And it also requires everybody to create their own metadata generator while there is one already [here](https://github.com/creativecommons/cc.engine/blob/cc74a5a/cc/engine/templates/chooser_pages/metadata.xmp).

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #197 by @fkohrt 

## Description

This PR adds the machine-readable code in .xmp format for embedding license information into files.

## Checklist
 
- [ ] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main` or `master`).
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by obulat** [comment]
@fkohrt, could you please look at this PR? I think you know a lot about xmp metadata.

File: src/utils/xmp.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    return data.replace(/ {2,}/gi, '').replace(/\n/gi, '')
+}
+
+export const createXMP = ({ shortName, workUrl = '', workTitle = '', creatorName = '', lang = 'en-US' }) => {
+    const slug = licenseSlug(shortName).replace(/-/gi, '_').toUpperCase()
+
+    const copyrighted = shortName !== LICENSES.CC0.SHORT
+    const xapRights = copyrighted
+        ? `<rdf:Description rdf:about='' xmlns:xapRights='http://ns.adobe.com/xap/1.0/rights/'>
+             <xapRights:Marked>true</xapRights:Marked></rdf:Description>`
```
Comment (fkohrt) [inline src/utils/xmp.js:13]: I believe the [XMP spec](https://wwwimages2.adobe.com/content/dam/acom/en/devnet/xmp/pdfs/XMP%20SDK%20Release%20cc-2016-08/XMPSpecificationPart1.pdf#G5.869889) says booleans are either exactly `True` or `False` (first char uppercase).

File: src/utils/xmp.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+${workTitle}</rdf:li><rdf:li xml:lang='${lang}'>${workTitle}</rdf:li></rdf:Alt></dc:title></rdf:Description>`
+        : ''
+
+    const licenseUrl = LICENSES[slug].URL
+
+    const ccLicenseUrl = `
+<rdf:Description rdf:about='' xmlns:cc='http://creativecommons.org/ns#'><cc:license rdf:resource='${licenseUrl}'/></rdf:Description>`
+
+    const ccLicenseNotice = `
+This work is licensed under <a rel="license noopener noreferrer" target="_blank" href="${licenseUrl}">${LICENSES[slug].FULL}</a>`
```
Comment (fkohrt) [inline src/utils/xmp.js:31]: I think HTML inside XML should be escaped, like "`This work is licensed under a &lt;a rel=&#34;license noopener noreferrer&#34; target=&#34;_blank&#34; href=&#34;${licenseUrl}&#34;&gt;${LICENSES[slug].FULL}&lt;/a&gt;`"...

**Review by fkohrt — CHANGES_REQUESTED** [review]
I looked at xmp.js and besides the two suggestions it looks good to me. I'd also suggest running a `diff` on the output of this vs. the [cc.engine](https://github.com/creativecommons/cc.engine/blob/cc74a5a/cc/engine/templates/chooser_pages/metadata.xmp).

File: src/utils/xmp.js (hunk 1 above)
Comment (obulat) [reply src/utils/xmp.js]: Correct, I haven't noticed that before. Thank you!

**Review by zackkrida — APPROVED** [review]
Love the `src/utils/xmp.js` helper. Nicely written code. Great!

**Pull request merged**